tests/test_load_wecdis.py

```diff
@@ -0,0 +1,637 @@
+
+    @staticmethod
+    def test_wecdis_parse_dza_only_parses_dza():
+        importer = WecdisImporter()
+        with pytest.raises(TypeError):
+            importer.handle_timestamp(DummyToken.csv_to_tokens("$POSL,XYZ,1234,NONSUCH*5D"), 1)
+        assert importer.timestamp is None
+
+    @staticmethod
+    def test_wecdis_parse_dza_invalid_line():
+        importer = WecdisImporter()
+        check_errors_for_file_contents(
+            "$POSL,DZA,20101030\nCHART, VER", "Not enough parts in line", importer
+        )
+
+    @staticmethod
+    def test_wecdis_parse_depth_invalid_value():
+        importer = WecdisImporter()
+        check_errors_for_file_contents(
+            "$POSL,PDS,U,M\nCHART,VER", "Couldn't convert to a number", importer
+        )
+
+    @staticmethod
+    def test_wecdis_parse_dza_invalid_timestamp():
+        importer = WecdisImporter()
+        check_errors_for_file_contents(
+            "$POSL,DZA,20101035,999999.99,12343*AB\nCHART, VER",
+            "Error in timestamp value",
+            importer,
+        )
+        assert importer.platform_name is None
+        assert importer.timestamp is None
+
+    def test_wecdis_parse_depth(self):
+        processor = FileProcessor(archive=False)
+        processor.register_importer(WecdisImporter())
+
+        # check states empty
+        with self.store.session_scope():
+            # there must be no states at the beginning
+            states = self.store.session.query(self.store.db_classes.State).all()
+            self.assertEqual(len(states), 0)
+
+            # there must be no platforms at the beginning
+            platforms = self.store.session.query(self.store.db_classes.Platform).all()
+            self.assertEqual(len(platforms), 0)
+
+            # there must be no datafiles at the beginning
+            datafiles = self.store.session.query(self.store.db_classes.Datafile).all()
+            self.assertEqual(len(datafiles), 0)
+
+        # parse the folder
+        processor.process(DEPTH_DATA_PATH, self.store, False)
+
+        # check data got created
+        with self.store.session_scope():
+            # there must be states after the import
+            states = self.store.session.query(self.store.db_classes.State).all()
+            self.assertEqual(len(states), 3)
+
+            # there must be platforms after the import
+            platforms = self.store.session.query(self.store.db_classes.Platform).all()
+            self.assertEqual(len(platforms), 1)
+
+            # there must be one datafile afterwards
+            datafiles = self.store.session.query(self.store.db_classes.Datafile).all()
+            self.assertEqual(len(datafiles), 1)
+
+            stored_states = self.store.session.query(self.store.db_classes.State).all()
+
+            ureg = UnitRegistry()
+            # 1 - subsurface
+            assert stored_states[0].elevation.to(ureg.meter).magnitude == -123
```

Comment:
This works fine, but possibly an easier way of doing it would to be do:

```
assert stored_states[0].elevation == -123 * ureg.meter
```

(In general, multiplying by a unit object assigns that unit)

Reply:
Yes, @mew-nsc - I'd like to start dog-fooding the `unknown platform` handling when I also test the WECDIS import against their data.

Are comfortable merging @robintw 's branch into this one @mew-nsc ?

Reply:
I'm happy with that. Does it make sense to change the target of the existing PR for the unknown platform handling to this branch, or to just do a straight merge across?